### PR TITLE
Added flag --listFns to list available functions, and relevant docume…

### PR DESCRIPTION
**Pull request by AlainLich**
Hi, nice and useful package. However found that I needed some guidance for using -z and -l flags.

So this PR adds flag `--listFns` to list available functions as a help for using 
   the `-z`  and `-l` flags. The argument value is a selector which is used
   for Unix pattern matching as with `-l` or `-z` flags; thereby helping the
   user select a subgraph. When arguments starts with `%`, all available information 
on selected entries is dumped.

    ~~~
    prof2dot.py -f pstats /tmp/myLog.profile  --listFns "test_segments:*:*" 
    
    test_segments:5:<module>,
    test_segments:206:TestSegments,
    test_segments:46:<lambda>


Doc. for proposed change in README-CHANGES.md (in https://github.com/AlainLich/gprof2dot branch ALMods)

### CODE REVIEW

**Comment by jrfonseca** [comment]
I think this is a good idea.

Just a couple of comments:
- option should be named `--list-functions` for consistency
- only the function names are used for `-z`/`-l` options (see Profile::getFunctionIds), so `printFunctionIds` should match it, and omit other bits of the function id (like line number module.)
- the `README-CHANGES.md` should be folded into  README, perhaps on a new subsection inside the Examples section

**Comment by AlainLich** [comment]
Hi, 
I will implement the suggestions and propose an updated PR, probably around the week end.

Concerning the 2nd remark, you mean match ignoring module and line info, and then list likewise (omitting same info) or
list the full entry? (or let the user specify?)

**Comment by jrfonseca** [comment]
Thanks. 

> Concerning the 2nd remark, you mean match ignoring module and line info, and then list likewise (omitting same info) or
list the full entry? (or let the user specify?)

Looking at `printFunctionIds` again, I think I misunderstood.  You're only dumping function names like the graph shows, so that's OK.

There's one other issue I noticed though: gprof2dot still supports Python 2, which doesn't support f-strings.

I should drop Python 2, and I'll do it eventually, but as far as this PR is concerned, I think it's simpler to just drop f strings, so that it doesn't need to wait.

**Comment by AlainLich** [comment]
Hi, 
I have implemented the changes you suggested, also made it run under Python2.7 and 3.3; tested in https://github.com/AlainLich/gprof2dot/actions/runs/889894702

1. Added a flag  `--max-acceptable` to the  `tests/test.py` program to make it more strict, ie reject after
   a threshold number of run errors has been attained. Default behaviour is compatible with previous: ie no rejection. 

2. Added 1 yaml file  `.github/workflows/testOnly.yml` to do testing only, used in above test, adjusts `--max-acceptable`  for strict test under Python2.7, allow for 2 
errors in Python3 (see below)

Bottom line
+ Python2: everything runs OK
+ Python3: the python examples fail, I guess that the test data file has been generated with Python2.7; test still succeeds with 
`--max-acceptable` set to 2

**Comment by AlainLich** [comment]
Hi,
Simple modifications in testing:
  - Added a test for the flag --function-list,
  - made differences in generated diff files count in test criterion. (in `diff` function gprof2dot around line 80)
  - results available in https://github.com/AlainLich/gprof2dot/actions/runs/893100611
Also, I had the (**may be not so good**) *idea* to delete-trailing-spaces in the Python file, so that it looks like many changes.

**Bottom line**: same as previous message, more tests performed and accounted for in the SUCCESS Status

1. Python2: everything runs OK
2. Python3: the python examples fail, I guess that the test data file has been generated with Python2.7; test still succeeds with
--max-acceptable set to 10 (2 tests generating dot, 7 tests for --function-list, 1 test for diff)

**Comment by jrfonseca** [comment]
Sorry for the delay.

Yep, its simpler to leave trailing whitespace alone.  Some people trim, others send PRs which contain them, and it ends up being noise in git history.  At any rate, I can't be bothered either way.

I think the issue with `--max-acceptable`, is that in some cases the gprof2dot output is not deterministic (it will depend on dictionary/set ordering.)  Anyway, it doesn't hurt to have the option.

**Comment by jrfonseca** [comment]
Thanks
